docs/DeveloperGuide.md

```diff
@@ -343,77 +343,208 @@ b. Require developers to download those libraries manually (this creates extra w
+`* *` | user with recurring tasks or events| add a task or an event that are reccuring| avoid the trouble of manually adding repeatedly on a regular basis
+`* *` | user | identify free time slots in a day, week or month | can evaluate how busy I am and optimize the date for tasks and events to add
+`* *` | user | display daily, weekly, or monthly calendar views | visualize the deadlines of tasks and times of events for the day/week/month
+`* *` | user | view the history of all previously undertaken actions | prevent duplicate commands and correct entries that are erroneous in hindsight
+`* *` | user | choose between automatic save and manual save |
+`*` | user | have an auto-completion for commands that are predictable | avoid the trouble of typing in the full command
+`*` | user | enter the options for a command in any order | 
+
+## Appendix B : Sample Use Cases
+
+(For all use cases below, the **System** is the `TaskCrusher` and the **Actor** is the `user`, unless specified otherwise)
+
+
+#### Use case: Add an event
+
+**MSS**
+
+1. User enters the name of an event to be added, together with its start and end date/time and other options.
+2. TaskCrusher adds the event to the active list
+3. TaskCrusher displays back the details of the add action.
+
+Use case ends.
+
+**Extensions**
 
-{More to be added}
+2a. The command format is invalid
 
-## Appendix B : Use Cases
+> 2a1. TaskCrusher shows an error message <br>
+  Use case resumes at step 1
+
+2b. The entered time slot is already occupied by another event 
+> 2b1. TaskCrusher notifies the user that the time slot for the event has been already occupied by other event.
+  2b2. User decides whether or not to force the addition. 
+  
+> Use case ends
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
 
-#### Use case: Delete person
+#### <a name = "list"> Use case: Display the active list
 
 **MSS**
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
-Use case ends.
+1. User requests to list active tasks, active events or both. Additional sort field may be specified.
+2. TaskCrusher displays the active list according to the listing options, assigning each entry an index.
+
+Use case ends
+
+**Extensions**
+
+2a. The command format is invalid
+
+> 2a1. TaskCrusher shows an error message <br>
+> Use case resumes at step 1
+
+2b. The active list is empty.
+> 2b1. TaskCrusher notifies the user that the active list is currently empty.<br/>
+> Use case ends
+
+#### Use case: Mark a task/event as `done`
+
+**MSS**
+
+1. User requests to [display the active list](#list).
+2. User uses the index of the task/event of interest to request to mark it as `done`. Multiple indices can be entered to process more than one entries. 
```

Comment:
system should display the list first, before user can tell what task to mark done